core/src/main/java/com/linecorp/armeria/client/Endpoint.java

```diff
@@ -418,21 +422,77 @@ public String authority() {
 
         if (isGroup()) {
             authority = "group:" + groupName;
-        } else if (port != 0) {
-            if (hostType == HostType.IPv6_ONLY) {
-                authority = '[' + host() + "]:" + port;
-            } else {
-                authority = host() + ':' + port;
-            }
-        } else if (hostType == HostType.IPv6_ONLY) {
-            authority = '[' + host() + ']';
         } else {
-            authority = host();
+            authority = buildAuthority(host(), port);
         }
 
         return this.authority = authority;
     }
 
+    /**
+     * Converts this endpoint into a URI using the {@code scheme}.
+     *
+     * @param scheme the {@code scheme} for {@link URI}.
+     *
+     * @return the URI
+     */
+    public URI toUri(String scheme) {
+        requireNonNull(scheme, "scheme");
+
+        return toUri(Scheme.parse(scheme));
+    }
+
+    /**
+     * Converts this endpoint into a URI using the {@link SessionProtocol} and {@link SerializationFormat}.
+     *
+     * @param sessionProtocol the {@link SessionProtocol} for {@link URI}.
+     * @param serializationFormat the {@link SerializationFormat} for {@link URI}
+     *
+     * @return the URI
+     */
+    public URI toUri(SessionProtocol sessionProtocol, SerializationFormat serializationFormat) {
+        requireNonNull(serializationFormat, "serializationFormat");
+        requireNonNull(sessionProtocol, "sessionProtocol");
+
+        return toUri(Scheme.of(serializationFormat, sessionProtocol));
+    }
+
+    /**
+     * Converts this endpoint into a URI using the {@link Scheme}.
+     *
+     * @param scheme the {@link Scheme} for {@link URI}.
+     *
+     * @return the URI
+     */
+    public URI toUri(Scheme scheme) {
+        requireNonNull(scheme, "scheme");
+
+        final String authority;
+        if (isGroup()) {
+            authority = "group:" + groupName;
+        } else {
+            authority = buildAuthority(hasIpAddr() ? ipAddr() : host(), port);
```

Comment:
Is it okay to use `ipAddr` instead of `host` from here? What if the scheme is `https`?

Reply:
I change to use `authority()` method when creating `URI`.